actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,17 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+
+      def extract_variant!
+        if parameters.key?(:format)
+          segments = parameters[:format].to_s.split("+")
```

Comment:
But this implementation doesn't allow url parameters including `:format` to have '+' characters. Actually, accessing `post.hello+world` raise routing error if route is `'post(.:msg)'`. Original behavior is not raising error, and get `msg: 'hello+world'. It may break back compatibility.